tests/integration/widgets/test_autocomplete_input.py

```diff
@@ -240,6 +243,42 @@ def test_case_sensitivity(self, bokeh_model_page) -> None:
 
         assert page.has_no_console_errors()
 
+    def test_strict(self, bokeh_model_page) -> None:
+        """Test effect of 'strict=False' with explicit JS callback"""
+        text_input = AutocompleteInput(css_classes=["foo"], completions = ["aAaBbb", "aAaBbB"], strict=False)
+        text_input.js_on_change('value', CustomJS(code=RECORD("value", "cb_obj.value")))
+
+        page = bokeh_model_page(text_input)
+
+        el = page.driver.find_element_by_css_selector('.foo input')
+        text = "not in completions"
+        enter_text_in_element(page.driver, el, text, click=1, enter=True)
+
+        results = page.results
+        assert results['value'] == text
+        assert page.has_no_console_errors()
+
+
+    def test_server_strict(self, bokeh_server_page) -> None:
+        """Test effect of 'strict=False' without explicit callback."""
+        text_input = AutocompleteInput(css_classes=["foo"], completions = ["aAaBbb", "aAaBbB"], strict=False)
+
+        def add_autocomplete(doc):
+            # note: for some reason, bokeh_server_page requires a 'canvas' in the document
```

Comment:
The previous tests do a further round trip update and check the value from a data source update via a `CustomAction` (on a plot)  In hindsight all the tests could probably be simplified to be more like this one.

Reply:
Should we open an issue for this?
Or would you like to see further changes in this PR?

Reply:
I think a discussion issue would be helpful, there's no need to do any of that work in this PR